chapters/ch04.md

```diff
@@ -68,10 +79,11 @@ Many other details such as neural network architecture are also important, and t
 [1] This diagram shows NNs doing better in the regime of small datasets. This effect is less consistent than the effect of NNs doing well in the regime of huge datasets. In the small data regime, depending on how the features are hand-engineered, traditional algorithms may or may not do better. For example, if you have 20 training examples, it might not matter much whether you use logistic regression or a neural network; the hand-engineering of features will have a bigger effect than the choice of algorithm. But if you have 1 million examples, I would favor the neural network.
 
 ->
-
+[1] Mặc dù hình vẽ cho thấy mạng neural cho kết quả tốt hơn với dữ liệu huấn luyện nhỏ, nhưng nhận định này ít đồng thuận hơn với nhận định về dùng mạng neural với dữ liệu huấn luyện lớn. Vì đối với dữ liệu nhỏ, chất lượng thuật toán truyền thống có thể hoặc không cho kết quả tốt hơn mạng neural và bị phụ thuộc rất nhiều vào cách tạo đặc trưng thủ công. Nếu ta chỉ có 20 mẫu huấn luyện thì không đủ để dùng cả hồi suy logistic lẫn mạng neural. Nhưng đồng thời, đặc trưng thủ công bị phụ thuộc nhiều vào thuật toán học máy. Còn nếu có một triệu mẫu, thì tôi sẽ chọn dùng mạng neural. 
 
 
 The process of how to accomplish (i) and (ii) are surprisingly complex. This book will discuss the details at length. We will start with general strategies that are useful for both traditional learning algorithms and neural networks, and build up to the most modern strategies for building deep learning systems.
 
 ->
+Quá trình hoàn thiện (i) huấn luyện mạng to hơn trên (ii) tập dữ liệu lớn hơn có thể phức tạp hơn bạn tưởng. Nhưng đừng lo, vấn đề này sẽ được thảo luận đầy đủ và chi tiết trong cuốn sách này. Nhưng trước tiên, chúng ta sẽ bắt đầu bằng các chiến thuật chung hữu ích cả cho thuật toán truyền thống lẫn mạng neural, và từ từ xây dựng chiến thuật mới nhất để xây dựng các hệ thống deep learning. 
```

Comment:
-> "Quá trình đạt được (i) một mạng lớn hơn và (ii) nhiều dữ liệu hơn phức tạp hơn chúng ta tưởng...."

Reply:
* "strategy" em nghĩ nên là chiến lược, cuốn sách này muốn nói đến chiến lược kiểu hoạch định từ đầu tới cuối làm những gì, còn chiến thuật thì hẹp hơn. Khi đã có chiến lược thì mới áp dụng chiến thuật để giải quyết vấn đề.
* "từ từ" may be là "và từng bước xây dựng lên"
* "the most modern" em nghĩ là không chỉ mới nhất mà còn hiện đại nhất (áp dụng những kỹ thuật mới nhất)

Reply:
- Chiến lược có vẻ hợp hơn (y)
< Chiến lượclà một tập hợp các lựa chọn được sử dụng để đạt được mục tiêu tổng thể trong khi chiến thuật là những hành động cụ thể được sử dụng khi áp dụng những lựa chọn chiến lược đó.
- Và từng bước xây dựng (không cần lên)
- the most modern --> ok.